Consolidate packaging properties

ViktorHofer opened the pull request:
The initial sweep of consolidating duplicated packaging properties. There is room for even more cleanup but that involves a bit more work which I will do in another PR.

ViktorHofer committed:
Consolidate packaging properties

Anipik commented on Directory.Build.props:
```diff
@@ -77,6 +77,11 @@
     <LicenseUrl>https://github.com/dotnet/runtime/blob/master/LICENSE.TXT</LicenseUrl>
     <RuntimeIdGraphDefinitionFile>$([MSBuild]::NormalizePath('$(LibrariesProjectRoot)', 'pkg', 'Microsoft.NETCore.Platforms', 'runtime.json'))</RuntimeIdGraphDefinitionFile>
     <LicenseFile>$(RepoRoot)LICENSE.TXT</LicenseFile>
+    <PackageLicenseFile>$(RepoRoot)LICENSE.TXT</PackageLicenseFile>
```
We have 2 properties LicenseFile on line 79 and packageLicenseFile. Can we just keep the LicenseFile ?
PackageLicenseFile is a deprecated property in nuget pack and cannot be set

Anipik commented on Directory.Build.props:
```diff
@@ -77,6 +77,11 @@
     <LicenseFile>$(RepoRoot)LICENSE.TXT</LicenseFile>
+    <PackageLicenseFile>$(RepoRoot)LICENSE.TXT</PackageLicenseFile>
+    <PackageLicenseExpression>MIT</PackageLicenseExpression>
+    <Copyright>$(CopyrightNetFoundation)</Copyright>
```
cant we keep these properties in packaging.props  as these are specific to packaging ? and just import that file everywhere ?

ViktorHofer replied:
eng/packaging.props and eng/packaging.targets contain libraries specifics that don't make sense to be imported everywhere. We already define some of the packaging properties in Directory.Build.props/targets so I don't see a problem with that.

ViktorHofer committed:
ThirdPartyNoticesFile and PackageLicenseFile

Anipik commented on eng/packaging.props:
```diff
@@ -49,17 +39,7 @@
 
-  <ItemGroup Condition="'$(MSBuildProjectExtension)' == '.pkgproj'">
-    <!-- Add required legal files to packages -->
-    <File Condition="Exists('$(PackageLicenseFile)')"
```
where are we adding the license file ?

ViktorHofer committed:
Add LicenseFile

safern commented on Directory.Build.targets:
```diff
@@ -22,6 +22,15 @@
 
   <!-- Packaging -->
+  <ItemGroup Condition="'$(IsPackable)' == 'true'">
+    <None Include="$(LicenseFile)"
```
Why `None` and not `File` with `SkipFileCheck=true`?

ViktorHofer replied:
`File` is an item that only the custom packaging system understands. `None` instead is supported by both  nuget and the custom packaging system. This will help @Anipik with this pkgproj conversion.

safern replied:
Ok cool.

safern commented on src/coreclr/dir.common.props:
```diff
@@ -41,15 +41,6 @@
-    <PackageVersion Condition="'$(PackageVersion)' == ''">6.0.0</PackageVersion>
-
-    <!-- Set the boolean below to true to generate packages with stabilized versions -->
-    <StabilizePackageVersion Condition="'$(StabilizePackageVersion)' == ''">false</StabilizePackageVersion>
```
These two properties are used for servicing and I will move them into the right location as a port of: https://github.com/dotnet/runtime/commit/2a51bcb2c1fb29b8ec8d8c4a30cb59e851446b00 to master

ViktorHofer replied:
I'm consolidating these properties across the whole repo. Are you saying that I shouldn't do that for coreclr?

ViktorHofer replied:
I won't merge this in before I've figured out the right fix that still allows to consolidate the properties. Will take another look at it tomorrow.

safern replied:
I'm saying for coreclr we don't need to consolidate them. These are already on Versions.props but coreclr needs to handle their servicing independently and that's why we need these for coreclr packages here.

ViktorHofer replied:
Curious, why does coreclr need to service independently than the rest of the repo?

safern replied:
because coreclr doesn't want their packages to be stable and shipped to NuGet on every servicing release. I think we discussed this here: https://github.com/dotnet/runtime/pull/45876#discussion_r540409002 in case that explanation helps here.

ViktorHofer replied:
I undid my changes for the sake of getting this in but I believe we should follow-up on a pattern that allows us to service all packages the same way.

safern commented on Directory.Build.targets:
```diff
@@ -22,6 +22,15 @@
   <!-- Packaging -->
+  <ItemGroup Condition="'$(IsPackable)' == 'true'">
+    <None Include="$(LicenseFile)"
+          PackagePath="$([System.IO.Path]::GetFileName('$(LicenseFile)'))"
```
If you leave `PackagePath` empty it puts it in the root right?

ViktorHofer replied:
`PackagePath` needs the file path supplied otherwise the custom packaging system throws:

```
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.20621.6\build\Packaging.targets(1203,5): error : System.Exception: Target path 'C:\git\runtime3\' contains invalid characters. [C:\git\runtime3\src\libraries\Microsoft.Bcl.AsyncInterfaces\pkg\Microsoft.Bcl.AsyncInterfaces.pkgproj]
```

ViktorHofer committed:
Consolidate the remaining pkg items

ViktorHofer committed:
Bring back StableVersion property

ViktorHofer committed:
Exclude license file for sfxprojs

ViktorHofer committed:
Remove versions.txt file from packages

The versions.txt file was added to packages to know which SHA a package
was built against. As the nuspec now contains the SHA, the versions file
isn't necessary anymore.

ViktorHofer committed:
More versions.txt cleanup

ViktorHofer committed:
One more versions.txt cleanup

ViktorHofer committed:
fixes

ViktorHofer committed:
Revert StableVersion changes

ViktorHofer committed:
one more revert
